lib/parser.dart

```diff
@@ -708,7 +708,7 @@ class _Parser {
 
         start = _peekToken.span;
         while (!_maybeEat(TokenKind.END_OF_FILE)) {
-          RuleSet ruleset = processRuleSet();
+          final ruleset = processRuleSet();
```

Comment:
Do you want to eliminate the type, shouldn't it be TreeNode ruleset (or maybe the name should change)?